asciidoctorj-core/src/main/java/org/asciidoctor/Options.java

```diff
@@ -67,10 +67,26 @@ public void setTemplateEngine(String templateEngine) {
         this.options.put(TEMPLATE_ENGINE, templateEngine);
     }
 
+    /**
+     * Enable writing output to a file. The file includes header and footer by default.
+     *
+     * @param toFile The path to the output file. If the path is not absolute, it is
+     *               interpreted relative to what was set via {@link #setToDir()}
+     *               or {@link #setBaseDir()} respectively.
+     *
+     */
     public void setToFile(String toFile) {
         this.options.put(TO_FILE, toFile);
     }
 
+    /**
+     * Toogle writing output to a file.
+     *
+     * @param toFile If <code>true</true>, overwrite the input file with the output,
```

Comment:
Would be great to have in here that this just changes the default of the headerFooter option and that it still can be changed.
Like this it reads that it's a fixed behavior creating an embeddable document with toFile=false and a full doc with toFile=true.